siliconcompiler/core.py

```diff
@@ -3967,10 +3981,19 @@ def run(self):
                     # we're not running with -resume, we also re-run anything
                     # in the steplist.
                     self.set('flowgraph', flow, step, index, 'status', None)
+
+                    # Reset metrics and records
+                    # We need to clear the 'set' flag to allow new metric values
+                    # to be merged in (since we merge manifests from each node
+                    # with clobber=False).
+                    # TODO: should there be a more formal way to do this? Or
+                    # should writing None do it implicitly?
```

Comment:
This is solving a different issue (which is that the `chip.set(...)` here will prevent the record from being set to the correct value later). However, if we change things so that setting a value to `None` effectively un-sets it like you mentioned in other comments, then this would be simplified.